operator/v1/types_ingress.go

```diff
@@ -211,6 +226,33 @@ type IngressControllerStatus struct {
 
 	// endpointPublishingStrategy is the actual strategy in use.
 	EndpointPublishingStrategy EndpointPublishingStrategy `json:"endpointPublishingStrategy"`
+
+	// conditions is a list of conditions and their status.
```

Comment:
There are a bunch of detail conditions but how do I know when the entire thing is ready, which as a consumer is all I care about?

Reply:
Oh, good point. Because we didn't have conditions before, I think we were leaning on `.status.availableReplicas` as a proxy for overall readiness, and now we have a way to publishing a much more meaningful condition. Will add that, thanks!

Reply:
Added an `Available` condition and docs.